portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -15,18 +15,33 @@
 package com.google.sps.servlets;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+@WebServlet("/random-fact")
 public class DataServlet extends HttpServlet {
 
+  private List<String> facts;
+
+  @Override
+  public void init() {
```

Comment:
When do you think would be a good time to use init() versus using a static list of strings?

Reply:
I based this code off of the walk-through code. From what I understand of Java this was done for the sake of **readability** and **maintainability**.

the "Array List" object is a vector an to add to it must be done with a function (in this case .add() ). 

The implementation of doing this in one line requires another function ( .asList() and .List.of())

Since there doesn't seem to be any way to make a list without Array List in Java doing so in the init seems like the best way **readability wise**.

**Maintainability wise** It doesn't seem to be best practice to run any constructing functions in the declaration of a class.

The init class also could be the first step to allowing for dependency injection so that the function can be tested with variation of string lists.

Reply:
So you would use init if you want to use dependency injection and/or add or delete from the list.

And you would make the list static if you don't want the list to be modified at all.

Making it static would also require you to make the instance of the list in one line or a block using the short hand for the .add() function.

 This form of instantiation can be done within the init() function making dependency injection possible for the static string.

Reply:
Yep, basically init() should only be used to change fields or implement logic that depend on an instance of the class. Otherwise, a static list is semantically clearest.

As a general rule, there's a strong preference to make variables private, final, and/or static whenever possible.